schema/crdb/fix-provisioning-counters/up.sql

```diff
@@ -0,0 +1,21 @@
+-- This change fixes provisioning counters, alongside the
+-- underflow fix provided in https://github.com/oxidecomputer/omicron/pull/5830.
+-- Although this underflow has been fixed, it could have resulted
+-- in invalid accounting, which is mitigated by this schema change.
+--
+-- This update is currently occurring offline, so we exploit
+-- that fact to identify that all instances should be off.
```

Comment:
nit: IMO the word "should" is doing more work than average here (there could be a Running instance if the servicing operator failed to stop it or hit something like #4511 that prevented them from stopping it). I think this is safe anyway for the reasons we discussed in chat (provisioning counter cleanup is idempotent now, so this won't impede other attempts to tell Nexus that the VMM is gone/drive the instance to Stopped), but I might leave a short note here to that effect.

Reply:
Sounds good, updated the comment here to elaborate on this point.